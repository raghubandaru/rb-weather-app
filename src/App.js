import React from "react";
import { connect } from "react-redux";
import Plotted from "./Plotted";
import Header from "./Header";
import {
  changeLocation,
  fetchData,
  setSelectedDate,
  setSelectedTemp
} from "./actions";
import "./App.css";

const API_KEY = process.env.REACT_APP_API_KEY;

export class App extends React.Component {
  fetchData = event => {
    event.preventDefault();
    let location = encodeURIComponent(this.props.redux.get("location"));
    let url = `http://api.openweathermap.org/data/2.5/forecast?q=${location}&APPID=${API_KEY}&units=metric`;

    this.props.dispatch(fetchData(url));
  };

  onPlotClick = data => {
    if (data.points) {
      var number = data.points[0].pointNumber;
      this.props.dispatch(
        setSelectedDate(this.props.redux.getIn(["dates", number]))
      );
      this.props.dispatch(
        setSelectedTemp(this.props.redux.getIn(["temps", number]))
      );
    }
  };

  changeLocation = event => {
    this.props.dispatch(changeLocation(event.target.value));
  };

  render() {
    let currentTemp = "Enter Location";
    if (this.props.redux.getIn(["data", "list"])) {
      currentTemp = this.props.redux.getIn([
        "data",
        "list",
        "0",
        "main",
        "temp"
      ]);
    }

    return (
      <div>
        <Header />
        <form onSubmit={this.fetchData}>
          <label>
            I want to know the weather for:
            <input
              type="text"
              placeholder={"City, Country"}
              value={this.props.redux.get("location")}
              onChange={this.changeLocation}
            />
          </label>
        </form>
        <div className="wrapper">
          <p className="temp-wrapper">
            <span className="temp">
              {this.props.redux.getIn(["selected", "temp"])
                ? this.props.redux.getIn(["selected", "temp"])
                : currentTemp}
            </span>
            {currentTemp !== "Enter Location" ? (
              <span className="temp-symbol">&deg;C</span>
            ) : (
              ""
            )}
            <span className="temp-date">
              {this.props.redux.getIn(["selected", "date"])
                ? this.props.redux.getIn(["selected", "date"])
                : ""}
            </span>
          </p>
          {this.props.redux.getIn(["data", "list"]) ? <h1>Forecast</h1> : null}
          {this.props.redux.getIn(["data", "list"]) ? (
            <Plotted
              xData={this.props.redux.get("dates")}
              yData={this.props.redux.get("temps")}
              onPlotClick={this.onPlotClick}
            />
          ) : null}
        </div>
      </div>
    );
  }
}

function mapStateToProps(state) {
  return {
    redux: state
  };
}

export default connect(mapStateToProps)(App);
