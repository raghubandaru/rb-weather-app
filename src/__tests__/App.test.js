import React from "react";
import { shallow } from "enzyme";
import { fromJS } from "immutable";
import { App } from "../App";

var redux = fromJS({
  location: "",
  data: {},
  dates: [],
  temps: [],
  selected: {
    date: "",
    temp: null
  }
});

describe("<App />", function() {
  it("renders correctly", function() {
    const wrapper = shallow(<App redux={redux} />);
    expect(wrapper).toMatchSnapshot();
  });
});
