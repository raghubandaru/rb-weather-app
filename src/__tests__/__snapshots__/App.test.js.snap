// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders correctly 1`] = `
<div>
  <Header />
  <form
    onSubmit={[Function]}
  >
    <label>
      I want to know the weather for:
      <input
        onChange={[Function]}
        placeholder="City, Country"
        type="text"
        value=""
      />
    </label>
  </form>
  <div
    className="wrapper"
  >
    <p
      className="temp-wrapper"
    >
      <span
        className="temp"
      >
        Enter Location
      </span>
      <span
        className="temp-date"
      />
    </p>
  </div>
</div>
`;
